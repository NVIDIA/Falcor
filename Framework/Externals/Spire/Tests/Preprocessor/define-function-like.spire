// support for function-like macros

#define FOO(x) 1.0 + x

float foo(float y) { return FOO(y) * 2.0; }

// simple token pasting

#define PASTE(a,b) a##b

PASTE(flo,at) bar() { return 0.0; }

// no space before parens? not a function-like macro

#define M (x) - (x)

// Error: undefined identifier `x`
float bar(float a) { return M(a); }
