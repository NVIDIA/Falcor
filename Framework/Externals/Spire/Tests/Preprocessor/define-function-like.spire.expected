result code = -1
standard error = {
Tests/Preprocessor/define-function-like.spire(15): error 30015: undefined identifier 'x'.
Tests/Preprocessor/define-function-like.spire(15): error 30015: undefined identifier 'x'.
}
standard output = {
}
